src/app/static/src/app/apiService.ts

```diff
@@ -70,9 +70,10 @@ export class APIService<S extends string, E extends string> implements API<S, E>
         return this;
     };
 
-    withError = (type: E, root = false) => {
+    withError = (type: E, root = false, payloadFunc?: (e: Error) => any) => {
         this._onError = (failure: Response) => {
-            this._commit({type: type, payload: APIService.getFirstErrorFromFailure(failure)}, {root});
+            const error = APIService.getFirstErrorFromFailure(failure);
+            this._commit({ type, payload: payloadFunc ? payloadFunc(error) : error }, { root });
```

Comment:
So this is what I had previously added `withErrorCallback` for. But I think what you've done is better. Could you remove the `withErrorCallback` helper and tidy up uses of that to use this instead?

But actually, I wonder if we should have 2 functions `withError` and `withErrorHandler`? Or some better name? Whatever you think!

Reply:
ohhhh damn sorry i didnt even see that, i was looking for that type of functionality, yh im happy to remove it, i think its nice to declare the mutation you want and all that, i personally think its nicer if we just have one function with an optional arg, it feels like exactly the same responsibility, just with pre commit step or not

Reply:
Nope that's fine, happy with 1 function if you want.